### PR TITLE
React UI: Point deletion context menu

### PR DESCRIPTION
**Pull request by ActiveChooN**
Delete point with simple context menu and without using `Ctrl+double click`
изображение

### CODE REVIEW

**Comment by bsekachev** [comment]
@ActiveChooN 

Hmm..
Whenever I try to press right button click on a point, UI fails with
```
Uncaught Error: Canvas is busy. Action: resize
    at CanvasModelImpl.activate (canvasModel.ts:336)
    at CanvasImpl.activate (canvas.ts:98)
    at CanvasWrapperComponent.activateOnCanvas (canvas-wrapper.tsx:498)
    at CanvasWrapperComponent.componentDidUpdate (canvas-wrapper.tsx:203)
    at commitLifeCycles (react-dom.development.js:22084)
    at commitLayoutEffects (react-dom.development.js:25302)
    at HTMLUnknownElement.callCallback (react-dom.development.js:336)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:385)
    at invokeGuardedCallback (react-dom.development.js:440)
    at commitRootImpl (react-dom.development.js:25040)
activate @ canvasModel.ts:336
activate @ canvas.ts:98
activateOnCanvas @ canvas-wrapper.tsx:498
componentDidUpdate @ canvas-wrapper.tsx:203
commitLifeCycles @ react-dom.development.js:22084
commitLayoutEffects @ react-dom.development.js:25302
callCallback @ react-dom.development.js:336
invokeGuardedCallbackDev @ react-dom.development.js:385
invokeGuardedCallback @ react-dom.development.js:440
commitRootImpl @ react-dom.development.js:25040
unstable_runWithPriority @ scheduler.development.js:818
runWithPriority$2 @ react-dom.development.js:12130
commitRoot @ react-dom.development.js:24889
finishSyncRender @ react-dom.development.js:24296
performSyncWorkOnRoot @ react-dom.development.js:24274
(anonymous) @ react-dom.development.js:12180
unstable_runWithPriority @ scheduler.development.js:818
runWithPriority$2 @ react-dom.development.js:12130
flushSyncCallbackQueueImpl @ react-dom.development.js:12175
flushSyncCallbackQueue @ react-dom.development.js:12163
batchedUpdates$1 @ react-dom.development.js:24359
notify @ Subscription.js:23
notifyNestedSubs @ Subscription.js:65
handleChangeWrapper @ Subscription.js:70
dispatch @ redux.js:221
(anonymous) @ redux-logger.js:1
(anonymous) @ index.js:11
onUpdateContextMenu @ canvas-wrapper.tsx:263
(anonymous) @ canvas-wrapper.tsx:473
contextmenuHandler @ canvasView.ts:465
react-dom.development.js:21810 The above error occurred in the <CanvasWrapperComponent> component:
    in CanvasWrapperComponent (created by ConnectFunction)
    in ConnectFunction (created by StandardWorkspaceComponent)
    in section (created by BasicLayout)
    in BasicLayout (created by Context.Consumer)
    in Adapter (created by StandardWorkspaceComponent)
    in StandardWorkspaceComponent (created by AnnotationPageComponent)
    in main (created by Basic)
    in Basic (created by Context.Consumer)
    in Adapter (created by AnnotationPageComponent)
    in section (created by BasicLayout)
    in BasicLayout (created by Context.Consumer)
    in Adapter (created by AnnotationPageComponent)
    in AnnotationPageComponent (created by ConnectFunction)
    in ConnectFunction (created by Context.Consumer)
    in withRouter(Connect(AnnotationPageComponent)) (created by Context.Consumer)
    in Route (created by CVATApplication)
    in Switch (created by CVATApplication)
    in GlobalHotKeys (created by CVATApplication)
    in main (created by Basic)
    in Basic (created by Context.Consumer)
    in Adapter (created by CVATApplication)
    in section (created by BasicLayout)
    in BasicLayout (created by Context.Consumer)
    in Adapter (created by CVATApplication)
    in CVATApplication (created by Context.Consumer)
    in withRouter(CVATApplication) (created by ConnectFunction)
    in ConnectFunction
    in Router (created by BrowserRouter)
    in BrowserRouter
    in Provider

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://fb.me/react-error-boundaries to learn more about error boundaries.
```
Can you confirm if it works for you?

**Comment by ActiveChooN** [comment]
> Can you confirm if it works for you?

Hmm, everything works fine for me, but i'll try to figure it out.

point_deletion

Also found that there is ability to open context menu for box points also. Will fix it.

**Comment by ActiveChooN** [comment]
@bsekachev please confirm that you still have an issue. And if so can you please provide a case in which it happens?

**Comment by bsekachev** [comment]
@ActiveChooN 

Was you able to reproduce it on ubuntu?

**Comment by ActiveChooN** [comment]
@bsekachev 
> 
> 
> @ActiveChooN
> 
> Was you able to reproduce it on ubuntu?

Yes, working on it

**Comment by bsekachev** [comment]
Some issues:
- When I open shape menu for a shape, then move cursor to another shape, I cannot open menu for another shape in a new place
- gif below
Peek 2020-03-23 15-02

Looks like when we remove some points of a shape, the shape updates immediately, when we remove others points of the shape, the shape doesn't update immediately. 

- one more issue demonstrated on the following gif
Peek 2020-03-23 15-07

- and I cannot remove the first point of a shape (a point that was first during drawing). Literally, I cannot even press the button 'Delete point'

File: cvat-ui/src/components/annotation-page/standard-workspace/canvas-wrapper.tsx [hunk 1]
```diff
@@ -4,20 +4,26 @@
...
-import { LogType } from 'cvat-logger';
-import { Canvas } from 'cvat-canvas';
-import getCore from 'cvat-core';
+import {
+    Layout,
+    Slider,
+    Icon,
+    Tooltip,
+} from 'antd';
+
```
Comment (bsekachev) [inline cvat-ui/src/components/annotation-page/standard-workspace/canvas-wrapper.tsx]: Let's import exact components in future code
Not from ``antd`` wrapper
It common recommendation to reduce bundle size

File: cvat-ui/src/components/annotation-page/standard-workspace/styles.scss [hunk 2]
```diff
@@ -119,3 +119,77 @@
...
+    position: fixed;
+    width: 300px;
+    z-index: 10;
+    max-height: 50%;
+    overflow-y: auto;
+
+    &:hover {
+        opacity: 1;
+    }
+}
```
Comment (bsekachev) [inline cvat-ui/src/components/annotation-page/standard-workspace/styles.scss]: Why have you duplicated style from ``src/components/annotation-page/styles.scss``?

File: cvat-ui/src/components/annotation-page/standard-workspace/styles.scss [hunk 3]
```diff
@@ -119,3 +119,77 @@
...
+
+        &:hover {
+            opacity: 1;
+        }
+
+        &:active {
+            opacity: 0.7;
+        }
+    }
+}
```
Comment (bsekachev) [inline cvat-ui/src/components/annotation-page/standard-workspace/styles.scss]: One more duplicate

File: cvat-ui/src/containers/annotation-page/standard-workspace/canvas-point-context-menu.tsx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+import { connect } from 'react-redux';
+import { CombinedState, ContextMenuType } from 'reducers/interfaces';
+
+import { updateAnnotationsAsync, updateCanvasContextMenu } from 'actions/annotation-actions';
+
+import CanvasPointContextMenuComponent from 'components/annotation-page/standard-workspace/canvas-point-context-menu';
+
+interface StateToProps {
+    activatedStateID: number | null;
+    activatedPointID: number | null | undefined;
```
Comment (bsekachev) [inline cvat-ui/src/containers/annotation-page/standard-workspace/canvas-point-context-menu.tsx]: ``activatedPointID?: number | null``

File: cvat-ui/src/containers/annotation-page/standard-workspace/canvas-point-context-menu.tsx [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+            dispatch(updateCanvasContextMenu(false, 0, 0));
+        },
+    };
+}
+
+type Props = StateToProps & DispatchToProps;
+
+interface State {
+    activatedStateID: number | null | undefined;
+    activatedPointID: number | null | undefined;
```
Comment (bsekachev) [inline cvat-ui/src/containers/annotation-page/standard-workspace/canvas-point-context-menu.tsx]: ``activatedPointID?: number | null``

File: cvat-ui/src/containers/annotation-page/standard-workspace/canvas-point-context-menu.tsx [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+    latestLeft: number;
+    latestTop: number;
+    left: number;
+    top: number;
+}
+
+class CanvasContextMenuContainer extends React.PureComponent<Props, State> {
```
Comment (bsekachev) [inline cvat-ui/src/containers/annotation-page/standard-workspace/canvas-point-context-menu.tsx]: ``CanvasPointContextMenuContainer``

**Pull request merged**